### PR TITLE
feat(pg|h3): support h3 functions

### PR DESCRIPTION
**Pull request by vdelacruzb**
# Description

Shortcut

- Story: https://app.shortcut.com/cartoteam/story/306889/implement-h3-functions-in-postgres
- Autolink: [sc-306889]

## Type of change

- Fix

# Acceptance

```
SELECT vdelacruz_carto.H3_BOUNDARY('847b59dffffffff');
-- POLYGON ((40.46506362234518 -3.9352772457964957, 40.546540602670504 -3.706115055436962, 40.387130040966305 -3.5142476508226355, 40.1459682881744 -3.551111388649416, 40.063938848332654 -3.7802950153411317, 40.22362190119583 -3.972594628037921, 40.46506362234518 -3.9352772457964957))
SELECT vdelacruz_carto.H3_CENTER('847b59dffffffff');
-- POINT (40.30547642317431 -3.743203325561684)
SELECT vdelacruz_carto.H3_DISTANCE('847b591ffffffff', '847b59bffffffff');
-- 1
SELECT vdelacruz_carto.H3_FROMGEOGPOINT(ST_POINT(40.4168, -3.7038), 4);
-- 847b59dffffffff
SELECT vdelacruz_carto.H3_FROMLONGLAT(40.4168, -3.7038, 4);
-- 847b59dffffffff
SELECT vdelacruz_carto.H3_HEXRING('837b59fffffffff', 1);
-- {837b5dfffffffff,837b58fffffffff,837b5bfffffffff,837a66fffffffff,837a64fffffffff,837b4afffffffff}
SELECT vdelacruz_carto.H3_ISPENTAGON('837b59fffffffff');
-- false
SELECT vdelacruz_carto.H3_ISPENTAGON('8075fffffffffff');
-- true
SELECT vdelacruz_carto.H3_ISVALID('847b59dffffffff');
-- true
SELECT vdelacruz_carto.H3_ISVALID('1');
-- false
SELECT vdelacruz_carto.H3_KRING('837b59fffffffff', 1);
-- {837b59fffffffff,837b58fffffffff,837b5bfffffffff,837a66fffffffff,837a64fffffffff,837b4afffffffff,837b5dfffffffff}
SELECT vdelacruz_carto.H3_POLYFILL(ST_GEOMFROMTEXT('POLYGON ((30 10, 40 40, 20 40, 10 20, 30 10))'), 4);
-- WORKS
SELECT vdelacruz_carto.H3_TOCHILDREN('837b59fffffffff', 4);
-- {847b591ffffffff,847b593ffffffff,847b595ffffffff,847b597ffffffff,847b599ffffffff,847b59bffffffff,847b59dffffffff}
SELECT vdelacruz_carto.H3_TOPARENT('847b59dffffffff', 3);
-- 837b59fffffffff
SELECT vdelacruz_carto.H3_RESOLUTION('847b59dffffffff');
-- 4
SELECT vdelacruz_carto.H3_COMPACT(ARRAY['857b59c3fffffff', '857b59c7fffffff', '857b59cbfffffff', '857b59cffffffff', '857b59d3fffffff', '857b59d7fffffff', '857b59dbfffffff']);
-- {847b59dffffffff}
SELECT vdelacruz_carto.H3_UNCOMPACT(ARRAY['847b59dffffffff'], 5);
-- {857b59c3fffffff,857b59c7fffffff,857b59cbfffffff,857b59cffffffff,857b59d3fffffff,857b59d7fffffff,857b59dbfffffff}
SELECT vdelacruz_carto.H3_KRING_DISTANCES('837b59fffffffff', 1);
-- WORKS
SELECT vdelacruz_carto.H3_INT_TOSTRING(596645165859340287);
-- 847b59dffffffff
SELECT vdelacruz_carto.H3_STRING_TOINT('847b59dffffffff');
-- 596645165859340287
```

### CODE REVIEW

File: clouds/postgres/common/build_modules.js [hunk 1]
```diff
@@ -129,16 +158,32 @@ function add (f, include) {
...
         });
     }
 }
 functions.forEach(f => add(f));
 
+let content = anonymousBlockWrapping(postgisInstalledCheck)
+
 // Replace environment variables
-let content = output.map(f => f.content).join('\n');
+content += output.map(f => anonymousBlockWrapping(extensionFuctionWrapping(f.content, f.module))).join('\n');
```
Comment (volaya) [inline clouds/postgres/common/build_modules.js:171]: Fuction => Function

**Review by volaya — APPROVED** [review]
Look good to me.

The only thing that I do not understand is why it uses the removeNextDuplicates function with the output of the H3 boundary. Shouldn't that already contain unique points (or at least not redundant points that are below the distance threshold)? If not, could we fix that in the JS library itself instead?

I might be missing something, though. In any case, it all looks fine. Good job!

**Comment by Jesus89** [comment]
This comes from an issue in Snowflake: https://github.com/CartoDB/analytics-toolbox-core/pull/371. The function `h3ToGeoBoundary` generates duplicated points in some cases.

It might be not a problem in Postgres, but I would keep it just in case.

File: clouds/postgres/modules/sql/h3/H3_RESOLUTION.sql [hunk 2]
```diff
@@ -0,0 +1,23 @@
+----------------------------
+-- Copyright (C) 2021 CARTO
```
Comment (Jesus89) [inline clouds/postgres/modules/sql/h3/H3_RESOLUTION.sql:2]: 2023

**Review by Jesus89 — APPROVED** [review]
Great job!

**Pull request merged**